Solutions/Branwen/GUI.cs

```diff
@@ -9,324 +9,460 @@
-                mySqlConnection.Open();
-
-                string deleteStatement = "TRUNCATE TABLE SaurutobiMediaPaths; TRUNCATE TABLE SaurutobiMediaFiles;";
-                MySqlCommand deleteCommand = new MySqlCommand(deleteStatement, mySqlConnection);
-                deleteCommand.ExecuteNonQuery();
-            }
-            catch (Exception ex)
-            {
-                MessageBox.Show("Failed to Wipe:" + Environment.NewLine + ex.Message);
-            }
-            buttonWipeDb.Enabled = true;
-            buttonSelectAndRunInventory.Enabled = true;
-            buttonSelectAndRunInventory.Text = "Select Inventory Directory";
-            textBoxMediaDriveNumber.Enabled = true;
-            UseDBCheckBox.Enabled = true;
-        }
-    }
+	public partial class GUI : Form
+	{
+		private int fileCount;
+
+		public GUI()
+		{
+			InitializeComponent();
+		}
+
+		#region Button Handlers
+
+		/// <summary>
+		/// Button Handler, Selects Directory then does Inventory
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="e"></param>
+		private void SelectAndRunInventoryButton_Click(object sender, EventArgs e)
+		{
+			fileCount = 0;
+			try
+			{
+				//make this thing and all associated stuff go away
+				SelectAndRunInventoryButton.Enabled = false;
+				SelectAndRunInventoryButton.Text = "Working";
+				WipeDbButton.Enabled = false;
+				WipeDbButton.Text = "...";
+				MediaDriveNumberTextBox.Enabled = false;
+				UseDBCheckBox.Enabled = false;
+				ExportFileCheckBox.Enabled = false;
+				FolderBrowserDialog folderDialog = new FolderBrowserDialog();
+				folderDialog.Description = "Set Folder to Inventory";
+				if (folderDialog.ShowDialog() != DialogResult.OK)
+				{
+					SelectAndRunInventoryButton.Enabled = true;
+					SelectAndRunInventoryButton.Text = "Select Inventory Directory";
+					UseDBCheckBox.Enabled = true;
+					SetEnabledControls();
+					return;
+				}
+
+				DirectoryInfo[] topLevelDirectories = new DirectoryInfo(folderDialog.SelectedPath).GetDirectories();
+				string outputFile = Path.Combine(folderDialog.SelectedPath, "MediadriveInventory.xlsx");
+				if (outputFile == null || topLevelDirectories == null)
+				{
+					MessageBox.Show("This Directory is Invalid. Please try again");
+					return;
+				}
+
+				if (UseDBCheckBox.Checked == true)
+				{
+					#region DataBase
+
+					MySqlConnection mySqlConnection;
+					try
+					{
+						mySqlConnection = new MySqlConnection("server=box654.bluehost.com;user=lbkstud1_smedia;password=#sucK_my_d1ck;database=lbkstud1_SaurutobiMedia;");
+						mySqlConnection.Open();
+						//Do the Delete
+						string deleteStatement = "DELETE FROM SaurutobiMediaPaths WHERE MediaDrive = " + MediaDriveNumberTextBox.Text + "; DELETE FROM SaurutobiMediaFiles WHERE MediaDrive = " + MediaDriveNumberTextBox.Text + ";";
+						MySqlCommand deleteCommand = new MySqlCommand(deleteStatement, mySqlConnection);
+						deleteCommand.ExecuteNonQuery();

```

Comment:
After this you should probably be closing your connection.  In fact, it might be easier to start the connection with a using statement so that it closes automatically at the end bracket of the using block.